File: src/_/Thunk.Tests/System/ObjectTests.cs
using Xunit;
namespace System
{
    /// <summary>
    /// ObjectTests
    /// </summary>
    public class ObjectTests
    {
        [Fact]
        public void Constructor_Valid_NotNull()
        {
            var obj = new object();
            Assert.NotNull(obj);
        }

        [Fact]
        public void Equals_OneObject_EqualsTrue()
        {
            var obj = new object();
            Assert.Equal(true, obj.Equals(obj));
        }

        [Fact]
        public void Equals2_OneObject_EqualsTrue()
        {
            var obj = new object();
            Assert.Equal(true, object.Equals(obj, obj));
        }

        [Fact]
        public void GetHashCode_OneObject_NotEqualsZero()
        {
            var obj = new object();
            Assert.NotEqual(0, obj.GetHashCode());
        }

        [Fact]
        public void GetType_OneObject_SameType()
        {
            var obj = new object();
            Assert.Same(obj.GetType(), typeof(object));
        }

        [Fact]
        public void ReferenceEquals_OneObject_EqualsTrue()
        {
            var obj = new object();
            Assert.Equal(true, object.ReferenceEquals(obj, obj));
        }

    }
}

File: src/_/Thunk.cpu.Tests/System/ObjectTests.cpp
using namespace System;
using namespace Xunit;

/*
///<summary>
/// ClassA class
///</summary>
public class(ClassA,System_Object,,
public char *Name;
);
Declare_DefaultCtor(public,ClassA)
Implement_Type(ClassA,System_Object,)
Implement_DefaultCtor(ClassA,System_Object)

///<summary>
/// ClassB class
///</summary>
public class(ClassB,ClassA,classInterface(System_IDisposable),
public char *Name;
);
Declare_DefaultCtor(public,ClassB)
Implement_Type(ClassB,ClassA,typeInterface(System_IDisposable))
Implement_DefaultCtor(ClassB,ClassA)
*/

namespace System
{
	/// <summary>
	/// ObjectTests
	/// </summary>
	public ref class ObjectTests
	{
	public:

		/// <summary>
		/// NewClass
		/// </summary>
		[Fact]
		void NewClass()
		{
			////char *name;
			//ClassA *classA = new(,ClassA);
			////name = vcall(System_Object,sample,GetHashCode);
			////printf("name: %d %s\n", name);
			//delete(classA);
		}

		/// <summary>
		/// VCall
		/// </summary>
		[Fact]
		void VCall()
		{
			////char *name;
			//ClassA *sample = new(,ClassA);
			////name = vcall(System_Object,sample,GetHashCode)((System_Object *)sample);
			////printf("name: %d %s\n", name);
			//delete(sample);
		}

		[Fact]
		void Constructor_Valid_NotNull()
		{
		}

		[Fact]
		void Equals_OneObject_EqualsTrue()
		{
		}

		[Fact]
		void Equals2_OneObject_EqualsTrue()
		{
		}

		[Fact]
		void GetHashCode_OneObject_NotEqualsZero()
		{
		}

		[Fact]
		void GetType_OneObject_SameType()
		{
		}

		[Fact]
		void ReferenceEquals_OneObject_EqualsTrue()
		{
		}
	};
}